### ISSUE TITLE
Query answer taking too much time

### ISSUE THREAD

**Issue by ma-garcia** [openlink/virtuoso-opensource#58]
I'm executing a query with the following structure:

PREFIX rdfs: http://www.w3.org/2000/01/rdf-schema#
PREFIx example: http://my.issue.example#
select ?item  
WHERE
{  
      ?item a example:certainType .
      ?item rdfs:label ?name .
      FILTER ( (regex(STR(?name),"^text1$"))||
                (regex(STR(?name),"^text2$"))||
                (regex(STR(?name),"^text3$"))||
                (regex(STR(?name),"^text4$"))||
                (regex(STR(?name),"^text5$"))||
                (regex(STR(?name),"^text6$"))||
                (regex(STR(?name),"^text7$"))||
                (regex(STR(?name),"^text8$"))||
                (regex(STR(?name),"^text9$"))
              )
}

This query works fine in Virtuoso 6, it takes little time. But if I run the same query, over the same data, in Virtuoso 7 (developer branch build May, 7, 2013), it seems to have no end while getting the results. No time error is displayed or any other kind of error. Could be that there is no index created for the labels?

**Comment by imitko** [comment]
please try with recent develop/7 also in case it still persist, at least send us result of explain () or give us sample data to  debug.

**Comment by ma-garcia** [comment]
As I'm not the owner of the data, I will have to consult with the owner if I can provide some partial data.

Meanwhile, I have tried the recent develop/7 build and this extension [1]. But if I try to run the following query:

SELECT ?label
WHERE
{
  ?x a ?b ; rdfs:label ?label . ?label bif:contains "'sant*'"
}

I get a timeout error. Any clue about this behaviour? I followed all the steps mentioned in [1] and I didn't get any error.

[1] http://docs.openlinksw.com/virtuoso/sparqlextensions.html
